Mathlib/LinearAlgebra/Dual/Basis.lean

```diff
@@ -240,7 +230,7 @@ elab "use_finite_instance" : tactic => evalUseFiniteInstance
 structure Module.DualBases (e : ι → M) (ε : ι → Dual R M) : Prop where
   eval_same : ∀ i, ε i (e i) = 1
   eval_of_ne : Pairwise fun i j ↦ ε i (e j) = 0
-  protected total : ∀ {m : M}, (∀ i, ε i m = 0) → m = 0
+  protected total : ∀ {m₁ m₂ : M}, (∀ i, ε i m₁ = ε i m₂) → m₁ = m₂
```

Comment:
What do you think of stating this with `Function.Injective (fun m => (ε · m))`?

Reply:
That might make it harder to prove?

Reply:
They're the same modulo `funext`, but maybe there's not much in it.